specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2024-11-01/AppServiceCertificateOrders.json

```diff
@@ -0,0 +1,1801 @@
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/CertificateOrderAction"
+              }
+            }
+          },
+          "default": {
+            "description": "App Service error response.",
+            "schema": {
+              "$ref": "../../../Microsoft.Web/stable/2024-11-01/CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Retrieve Certificate Order Actions": {
+            "$ref": "./examples/RetrieveCertificateOrderActions.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CertificateRegistration/certificateOrders/{name}/retrieveEmailHistory": {
+      "post": {
+        "tags": [
+          "AppServiceCertificateOrders"
+        ],
+        "summary": "Retrieve email history.",
+        "description": "Description for Retrieve email history.",
+        "operationId": "AppServiceCertificateOrders_RetrieveCertificateEmailHistory",
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "name": "name",
+            "in": "path",
+            "description": "Name of the certificate order.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/CertificateEmail"
+              }
+            }
+          },
+          "default": {
+            "description": "App Service error response.",
+            "schema": {
+              "$ref": "../../../Microsoft.Web/stable/2024-11-01/CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Retrieve Certificate Email History": {
+            "$ref": "./examples/RetrieveCertificateEmailHistory.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "AppServiceCertificate": {
+      "description": "Key Vault container for a certificate that is purchased through Azure.",
+      "type": "object",
+      "properties": {
+        "keyVaultId": {
+          "description": "Key Vault resource Id.",
+          "type": "string"
+        },
+        "keyVaultSecretName": {
+          "description": "Key Vault secret name.",
+          "type": "string"
+        },
+        "provisioningState": {
```

Comment:
>provisioningState [](http://example.com/codeflow?start=9&length=17)

The provisioningState property has specific meaning to ARM, and is required to only contain 3 terminal states: Succeeded, Failed, and Canceled. Additional states are fine, but those should all be transition states. If you need to map these states, you should choose a different property name, e.g. secretProvisioningState, secretStatus, or something else you like better.